app/services/combined_audio_derivative_creator.rb

```diff
@@ -0,0 +1,163 @@
+require 'tempfile'
+#
+# Generate the combined audio derivatives for a
+# given oral history work.
+#
+# some_work = Work.find_by_friendlier_id(friendlier_id)
+# CombinedAudioDerivativeCreator.new(some_work).generate
+#
+#
+# Sample output:
+#     {
+#      :combined_audio_mp3_data=>
+#          "/var/folders/n7/t4zt2w751bj_6pbnf95sgpnrmz8lz8/T/output20200228-61637-ec6kjz.mp3",
+#      :combined_audio_webm_data=>
+#          "/var/folders/n7/t4zt2w751bj_6pbnf95sgpnrmz8lz8/T/output20200228-61637-1btjscz.webm",
+#      :fingerprint=>"4998d366f5edb6222db1181c7b153e21",
+#      :component_metadata=>
+#        {:durations=>
+#          [
+#            "00:30:30.29",
+#            "00:29:13.32",
+#            "00:33:03.26",
+#            "00:24:28.94"
+#          ]
+#        }
+#     }
```

Comment:
Returning as a hash is kind of messy, it's easy for the consumer to typo a key and miss the thing. 

How about making a very simple response object to return instead? Here's one easy way to make a nested response class:

```ruby
Response = Struct.new(:webm_file, :mp3_file, :fingerprint, keyword_init: true)

# then you can do
Response.new(webm_file: whatever)
# and also
r = Response.new
r.webm_file = whatever
r.webm_file
# etc
```

Reply:
Good call. Followed your advice.